examples/mujoco/ppo/train_ppo.py

```diff
@@ -0,0 +1,199 @@
+                        help='Level of the root logger.')
+    parser.add_argument('--monitor', action='store_true',
+                        help='Wrap env with gym.wrappers.Monitor.')
+    parser.add_argument('--log-interval', type=int, default=1000,
+                        help='Interval in timesteps between outputting log'
+                             ' messages during training')
+    parser.add_argument('--update-interval', type=int, default=2048,
+                        help='Interval in timesteps between model updates.')
+    parser.add_argument('--epochs', type=int, default=10,
+                        help='Number of epochs to update model for per PPO'
+                             ' iteration.')
+    parser.add_argument('--batch-size', type=int, default=64,
+                        help='Minibatch size')
+    args = parser.parse_args()
+
+    logging.basicConfig(level=args.logger_level)
+
+    # Set a random seed used in ChainerRL
+    misc.set_random_seed(args.seed, gpus=(args.gpu,))
+
+    # Set different random seeds for different subprocesses.
+    # If seed=0 and processes=4, subprocess seeds are [0, 1, 2, 3].
+    # If seed=1 and processes=4, subprocess seeds are [4, 5, 6, 7].
+    process_seeds = np.arange(args.num_envs) + args.seed * args.num_envs
+    assert process_seeds.max() < 2 ** 32
+
+    args.outdir = experiments.prepare_output_dir(args, args.outdir)
+
+    def make_env(process_idx, test):
+        env = gym.make(args.env)
+        # Use different random seeds for train and test envs
+        process_seed = int(process_seeds[process_idx])
+        env_seed = 2 ** 32 - 1 - process_seed if test else process_seed
+        env.seed(env_seed)
+        # Cast observations to float32 because our model uses float32
+        env = chainerrl.wrappers.CastObservationToFloat32(env)
+        if args.monitor:
+            env = gym.wrappers.Monitor(env, args.outdir)
+        if args.render:
+            env = chainerrl.wrappers.Render(env)
+        return env
+
+    def make_batch_env(test):
+        return chainerrl.envs.MultiprocessVectorEnv(
+            [functools.partial(make_env, idx, test)
+             for idx, env in enumerate(range(args.num_envs))])
+
+    # Only for getting timesteps, and obs-action spaces
+    sample_env = gym.make(args.env)
+    timestep_limit = sample_env.spec.tags.get(
+        'wrapper_config.TimeLimit.max_episode_steps')
+    obs_space = sample_env.observation_space
+    action_space = sample_env.action_space
+    print('Observation space:', obs_space)
+    print('Action space:', action_space)
+
+    assert isinstance(action_space, gym.spaces.Box)
+
+    # Normalize observations based on their empirical mean and variance
+    obs_normalizer = chainerrl.links.EmpiricalNormalization(
+        obs_space.low.size, clip_threshold=5)
+
+    # Orthogonal weight initialization is used as OpenAI Baselines does
+    winit = chainerrl.initializers.Orthogonal(1.)
```

Comment:
These scale doesn't seem to be the same as the one used in the baseline's repo: https://github.com/openai/baselines/blob/master/baselines/common/models.py#L96
The other one seems okay though:
https://github.com/openai/baselines/blob/master/baselines/common/policies.py#L49

Reply:
Nevermind about the scale, this one seems to agree with:
https://github.com/Breakend/baselines/blob/50ffe01d254221db75cdb5c2ba0ab51a6da06b0a/baselines/ppo1/mlp_policy.py#L33

However the initializer appears to not be the same:
https://github.com/Breakend/baselines/blob/50ffe01d254221db75cdb5c2ba0ab51a6da06b0a/baselines/ppo1/mlp_policy.py#L28
https://github.com/Breakend/baselines/blob/50ffe01d254221db75cdb5c2ba0ab51a6da06b0a/baselines/common/tf_util.py#L294

Reply:
You are right, it seems they actually did not use orthogonal weight initialization at that time. I clarified it in the comment and README.

Reply:
Ok, everything else looks good to me